core/src/main/java/com/spotify/metrics/core/SemanticMetricBuilderFactory.java

```diff
@@ -0,0 +1,47 @@
+package com.spotify.metrics.core;
+
+import com.codahale.metrics.Histogram;
+import com.codahale.metrics.Metric;
+import com.codahale.metrics.Reservoir;
+import com.codahale.metrics.Timer;
+import java.lang.reflect.InvocationTargetException;
+import java.util.function.Supplier;
+
+public class SemanticMetricBuilderFactory {
+
+    public static SemanticMetricBuilder<Timer> timerWithReservoir(
+        final Supplier<Reservoir> reservoirSupplier) {
+        return reservoir(Timer.class, reservoirSupplier);
+    }
+
+    public static SemanticMetricBuilder<Histogram> histogramWithReservoir(
+        final Supplier<Reservoir> reservoirSupplier) {
+        return reservoir(Histogram.class, reservoirSupplier);
+    }
+
+    private static <T extends Metric> SemanticMetricBuilder<T>  reservoir(Class<T> klazz,
```

Comment:
This reduces some code duplication but adds lots of ugly error handling. I'm not sure I like it. Since it is private we have control over what its called as, so it _shouldnt_ run into these errors, but who knows.